src/libstore/local-store.cc

```diff
@@ -529,9 +529,16 @@ static void canonicalisePathMetaData_(const Path & path, uid_t fromUid, InodesSe
             throw SysError("querying extended attributes of '%s'", path);
 
         for (auto & eaName: tokenizeString<Strings>(std::string(eaBuf.data(), eaSize), std::string("\000", 1))) {
-            /* Ignore SELinux security labels since these cannot be
-               removed even by root. */
-            if (eaName == "security.selinux") continue;
+            bool ignore =  false;
+            std::list<std::string>::iterator it;
+            std::list<std::string> ignoredAcls = settings.ignoredAcls.get();
+            for (it = ignoredAcls.begin(); it != ignoredAcls.end(); ++it) {
+                if (eaName == *it) {
+                    ignore = true;
+                    break;
+                }
+            }
+            if (ignore) continue;
```

Comment:
If you change the type of `settings.ignoredAcls` to `StringSet`, then it should be possible to write this as
```
if (settings.ignoredAcls.get().count(eaName)) continue;
```